### PR TITLE
Use ``python-typing-update`` on half of the ``tests`` directory

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Ref https://github.com/PyCQA/pylint/issues/4683.

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]

File: tests/test_func.py [hunk 1]
```diff
@@ -4,17 +4,18 @@
...
+from os.path import dirname, join
 
 import pytest
 
 from pylint.testutils import UPDATE_FILE, UPDATE_OPTION, _get_tests_info, linter
 
-INPUT_DIR = join(dirname(abspath(__file__)), "input")
-MSG_DIR = join(dirname(abspath(__file__)), "messages")
+INPUT_DIR = join(dirname(__file__), "input")
+MSG_DIR = join(dirname(__file__), "messages")
```
[the remark this answers is not in the document]
Comment (Pierre-Sassoulas) [reply tests/test_func.py:18]: Hmm that's suspicious though, in python 3.8 ``__file__`` is not abspathed:

```
cat a.py    
print(__file__)
python3 a.py
a.py
```
```
cat a.py 
import os

print(os.path.abspath(__file__))
python3 a.py
/home/pierre/astroid/a.py
```

[the remark this answers is not in the document]
Comment (DanielNoord) [reply tests/test_func.py:18]: Hmm is it related to `dirname` perhaps?

[the remark this answers is not in the document]
Comment (Pierre-Sassoulas) [reply tests/test_func.py:18]: Ho, it's actually worst if we take dirname into account:
```
cat a.py    
import os

print(os.path.dirname(__file__))
python3 a.py
 #nothing printed
```
```
cat a.py    
import os

print(os.path.dirname(os.path.abspath(__file__)))
 pierre@pierre-VirtualBox  ~/astroid   dependabot/pip/pylint-approx-eq-2.13.5  python3 a.py
/home/pierre/astroid
```
(This is python 3.8)

[the remark this answers is not in the document]
Comment (DanielNoord) [reply tests/test_func.py:18]: Do we want to revert?

[the remark this answers is not in the document]
Comment (Pierre-Sassoulas) [reply tests/test_func.py:18]: I think it's safer.

[the remark this answers is not in the document]
Comment (cdce8p) [reply tests/test_func.py:18]: > I think it's safer.

I agree. Let's revert it here.

File: tests/test_pylint_runners.py [hunk 2]
```diff
@@ -18,7 +19,7 @@
     "runner", [run_epylint, run_pylint, run_pyreverse, run_symilar]
 )
 def test_runner(runner: Callable, tmpdir: LocalPath) -> None:
-    filepath = os.path.abspath(__file__)
+    filepath = __file__
```
Comment (cdce8p) [inline tests/test_pylint_runners.py:22]:
```suggestion
    filepath = os.path.abspath(__file__)
```
Let's focus just on the typing change here.

File: tests/test_pylint_runners.py [hunk 3]
```diff
@@ -32,7 +33,7 @@ def test_runner(runner: Callable, tmpdir: LocalPath) -> None:
 )
 def test_runner_with_arguments(runner: Callable, tmpdir: LocalPath) -> None:
     """Check the runners with arguments as parameter instead of sys.argv."""
-    filepath = os.path.abspath(__file__)
+    filepath = __file__
```
Comment (cdce8p) [inline tests/test_pylint_runners.py:36]:
```suggestion
    filepath = os.path.abspath(__file__)
```

File: tests/test_regr.py [hunk 4]
```diff
@@ -5,20 +5,24 @@
...
+from typing import cast
 
 import astroid
 import pytest
 
 from pylint import testutils
 from pylint.lint.pylinter import PyLinter
 
-REGR_DATA = join(dirname(abspath(__file__)), "regrtest_data")
+REGR_DATA = join(dirname(__file__), "regrtest_data")
```
Comment (cdce8p) [inline tests/test_regr.py:25]:
```suggestion
REGR_DATA = join(dirname(abspath(__file__)), "regrtest_data")
```

File: tests/test_func.py (hunk 1 above)
Comment (cdce8p) [inline tests/test_func.py:18]:
```suggestion
INPUT_DIR = join(dirname(abspath(__file__)), "input")
MSG_DIR = join(dirname(abspath(__file__)), "messages")
```

**Review by cdce8p — CHANGES_REQUESTED** [review]
Let's just focus on the typing changes here.
As part of `python-typing-update`, `pyupgrade` is run. The version setting is passed directly to it. If those updates depend on `--py39-plus` we probably shouldn't do them at this time.

**Review by cdce8p — APPROVED** [review]

**Pull request merged**